src/briefcase/platforms/web/static.py

```diff
@@ -0,0 +1,347 @@
+            "--no-browser",
+            action="store_false",
+            dest="open_browser",
+            help="Don't open a web browser on the newly opened server.",
+            required=False,
+        )
+
+    def run_app(self, app: AppConfig, host, port, open_browser, **kwargs):
+        """Start the application.
+
+        :param app: The config object for the app
+        :param host: The host on which to run the server
+        :param port: The device UDID to target. If ``None``, the user will
+            be asked to select a device at runtime.
+        :param open_browser: Should a browser be opened on the newly started
+            server.
+        """
+        self.logger.info(
+            f"Starting web server on http://{host}:{port}/",
+            prefix=app.app_name,
+        )
+
+        httpd = None
+        try:
+            # Create a local HTTP server
+            httpd = LocalHTTPServer(
+                self.project_path(app),
+                host=host,
+                port=port,
+            )
+
+            # If requested, open a brower tab on the newly opened server.
+            if open_browser:
+                webbrowser.open_new_tab(f"http://{host}:{port}")
+
+            self.logger.info(
+                "Web server log output (type CTRL-C to stop log)...",
+                prefix=app.app_name,
+            )
+            self.logger.info("=" * 75)
+
+            # Run the server.
+            httpd.serve_forever()
+        except KeyboardInterrupt:
+            # CTRL-C is the accepted way to stop the server.
+            pass
+        finally:
+            if httpd:
+                with self.input.wait_bar("Shutting down server..."):
+                    httpd.server_close()
+
+            # Not sure why, but this is needed to mollify coverage for the
+            # "test_cleanup_server_error" case. Without this pass, a missing branch
+            # is reported for the "if httpd: -> exit" branch
+            pass
+
+        return {}
+
+
+class StaticWebPackageCommand(StaticWebMixin, PackageCommand):
+    description = "Package a static web app."
+
+    @property
+    def packaging_formats(self):
+        return ["zip"]
+
+    @property
+    def default_packaging_format(self):
+        return "zip"
+
+    def package_app(self, app: AppConfig, packaging_format: str, **kwargs):
+        """Package an app for distribution.
+
+        :param app: The app to package.
+        :param packaging_format: The packaging format; the only supported value
+            is ``zip``, enforced at the command level.
+        """
+        self.logger.info(
+            "Packaging web app for distribution...",
+            prefix=app.app_name,
+        )
+
+        with self.input.wait_bar("Building archive..."):
+            with ZipFile(
+                self.distribution_path(app, packaging_format=packaging_format), "w"
+            ) as archive:
+
+                for filename in self.project_path(app).glob("**/*"):
```

Comment:
Good opportunity for a progress bar....but using the glob generator is obviously more efficient.

Reply:
Good point about possibly using a progress bar; however, my experience is that this archiving is sufficiently fast that progress won't add *that* much. Given the complexity of needing to pre-process the glob, I'm inclined to keep this as a waitbar for now.